include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -76,19 +76,40 @@ class MPMCBoundedQueue
     MPMCBoundedQueue& operator=(MPMCBoundedQueue&& rhs) noexcept;
 
     /**
-     * @brief push Push data to queue.
+     * @brief pushStrong Push data to queue.
      * @param data Data to be pushed.
-     * @return true on success.
+     * @return true on success, if the result is false it is safe to 
+     * infer that the queue is full at the time of the call.
      */
     template <typename U>
-    bool push(U&& data);
+    bool pushStrong(U&& data);
```

Comment:
I dichotomized the push/pop interface such that it is similar to `atomic::compare_exchange_weak`/`atomic::compare_exchange_strong` - the weak versions are faster, but may fail spuriously.